src/rgw/driver/rados/rgw_sync.cc

```diff
@@ -1927,6 +1988,7 @@ class RGWMetaSyncCR : public RGWCoroutine {
   using ControlCRRef = boost::intrusive_ptr<RGWMetaSyncShardControlCR>;
   using StackRef = boost::intrusive_ptr<RGWCoroutinesStack>;
   using RefPair = std::pair<ControlCRRef, StackRef>;
+  RGWCoroutinesStack *notify_stack;
```

Comment:
can use `StackRef` instead of a raw pointer to make sure the stack doesn't get deleted early; i think it would if `RGWMetaSyncShardNotifyCR` hit its error case

Reply:
It wasn't very clear if I should use StackRef with RGWMetaSyncShardNotifyCR(), the way it has been paired with ControlCRRef? I have simply used an intrusive_ptr instead of raw pointer.  is that ok?